src/libstd/sys/windows/fs.rs

```diff
@@ -377,8 +377,11 @@ impl fmt::Debug for File {
     }
 }
 
-pub fn to_utf16(s: &Path) -> Vec<u16> {
-    s.as_os_str().encode_wide().chain(Some(0)).collect()
+pub fn to_u16s(s: &Path) -> io::Result<Vec<u16>> {
+    if s.as_os_str().to_inner().to_inner().iter().any(|&b| b == 0) {
```

Comment:
While that may be true, yeah, it's more robust to check what we're actually returning rather than whatever encoding `OsString` happens to be in.